cpp/benchmarks/join/generate_input_tables.cuh

```diff
@@ -0,0 +1,408 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#ifndef __GENERATE_INPUT_TABLES_CUH
+#define __GENERATE_INPUT_TABLES_CUH
+
+#include <curand.h>
+#include <curand_kernel.h>
+#include <thrust/sequence.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/distance.h>
+#include <cassert>
+
+#include <cudf/cudf.h>
+#include <rmm/rmm.h>
+
+#include "error.cuh"
+
+
+__device__ __inline__
+int64_t atomicCAS(int64_t* address, int64_t compare, int64_t val)
+{
+    return (int64_t)atomicCAS((unsigned long long int*)address, (unsigned long long int)compare, (unsigned long long int)val);
+}
+
+
+template <typename col_type>
+gdf_dtype gdf_dtype_from_col_type()
```

Comment:
If you need this functionality, use `gdf_dtype_of<T>` in `src/utililities/type_dispatcher.hpp`